find_path.py

```diff
@@ -130,11 +139,13 @@ def calc_cosine_sim(emb,path_nodes,g_nodes,triples_df,search_type,labels_all,kg_
     for l in path_nodes:
         cs = 0
         for i in range(0,len(l)-1):
-            if l[i] not in list(embeddings.keys()):
-                e = get_embedding(emb,l[i])
-                embeddings[l[i]] = e
+            n1 = g_nodes[l[i]]
+            n1_int = convert_path_nodes(n1,entity_map)
+            if n1_int not in list(embeddings.keys()):
```

Comment:
I'm confused here. I think this is the crux of the issue, but I don't understand why the node key would not be in the embeddings if they are being calculated correctly. I'll need to test

Reply:
Also, this code is currently for cosine similarity, but we may want to use the embeddings for other distance metrics or functionality (e.g., link prediction). Just a thought.

Reply:
I'll need to test tomorrow. Will approve then.